### DISCUSSION TITLE
GraphQL config not recognized by graphql-eslint

### DISCUSSION THREAD

**Discussion by FloEdelmann** [graphql-hive/graphql-eslint#2803] [Q&A] [answered]
I started migrating to graphql-eslint v4 now. My minimal ESLint configuration (`.eslintrc.js`) looks like this:

```js
module.exports = {
    root: true,
    overrides: [
        {
            files: '**/*.graphql',
            extends: 'plugin:@graphql-eslint/operations-recommended',
        },
    ],
}
```

There is a `graphql.config.ts` file in the same folder. The GraphQL files are located in subdirectories. Linting yields this error:

```
Error while loading rule '@graphql-eslint/no-deprecated': Rule `no-deprecated` requires graphql-config `schema` field to be set and loaded. See https://the-guild.dev/graphql/eslint/docs/usage#providing-schema for more info
```

I investigated a bit by adding `console.log` statements to various functions in graphql-eslint and graphql-config:

* [`loadOnDiskGraphQLConfig`](https://github.com/dimaMachina/graphql-eslint/blob/d455b5be393fdc4b1afdf83f66c572681036196e/packages/plugin/src/graphql-config.ts#L22) is called with the file path of the GraphQL file in the subdirectory.
* [`getFirstExistingPath(path.dirname(filePath))`](https://github.com/dimaMachina/graphql-eslint/blob/d455b5be393fdc4b1afdf83f66c572681036196e/packages/plugin/src/graphql-config.ts#L25C14-L25C58) thus returns that subdirectory.
* Since that subdirectory doesn't include the GraphQL config file, [`loadConfigSync`](https://github.com/dimaMachina/graphql-eslint/blob/d455b5be393fdc4b1afdf83f66c572681036196e/packages/plugin/src/graphql-config.ts#L23) returns `undefined`.
* If I manually pass the root directory path as `rootDir` instead, linting works as expected.

So I think that the logic for finding the nearest GraphQL config file is wrong, since it doesn't traverse up the directory tree (until the working directory, or until the directory that the ESLint config file is in), or do I miss something?

Also, the location of the GraphQL config file can't be specified, so there is no way around the issue, right?

**Comment by dimaMachina** [comment]
Hi, please try to make this test fail https://github.com/dimaMachina/graphql-eslint/blob/d455b5be393fdc4b1afdf83f66c572681036196e/packages/plugin/__tests__/schema.spec.ts#L115

**Reply by FloEdelmann** [reply, marked as answer]
I tried but couldn't. But thanks for the hint anyway.

In the end I updated the (transitive) `graphql-config` dependency from 5.1.2 to 5.1.3 and that fixed it :partying_face:

**Reply by FloEdelmann** [reply]
The https://github.com/kamilkisiela/graphql-config/releases/tag/v5.1.3 release notes mention only [one PR](https://github.com/kamilkisiela/graphql-config/pull/1572) and that fixes [the exact issue](https://github.com/kamilkisiela/graphql-config/issues/1506) I had.

**Reply by dimaMachina** [reply]
seems we need update a range in https://github.com/dimaMachina/graphql-eslint/blob/d455b5be393fdc4b1afdf83f66c572681036196e/packages/plugin/package.json#L53, would you want to submit a PR?

**Reply by FloEdelmann** [reply]
Ah indeed, that prevents this problem for other users. I created a PR: #2805
